File: problems/1147_longest_chunked_palindrome_decomposition/solution.py
class Solution:
    def longestDecomposition(self, text: str) -> int:
        n = len(text)
        result = 0
        left = 0
        right = n - 1
        l_str = ''
        r_str = ''
        while left < right:
            l_str += text[left]
            r_str = text[right] + r_str
            if l_str == r_str:
                result += 2
                l_str = ''
                r_str = ''
            left += 1
            right -= 1
        if left == right or l_str:
            result += 1
        return result

File: problems/1147_longest_chunked_palindrome_decomposition/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int longestDecomposition(string text) {
        int n = text.size();
        int result = 0;
        string lStr, rStr;
        int left = 0, right = n - 1;
        while (left < right) {
            lStr += text[left];
            rStr = text[right] + rStr;
            if (lStr == rStr) {
                result += 2;
                lStr = "";
                rStr = "";
            }
            left++;
            right--;
        }
        if (left == right || !lStr.empty()) result++;
        return result;
    }
};
